pkg/container-collection/match.go

```diff
@@ -29,6 +29,9 @@ func ContainerSelectorMatches(s *ContainerSelector, c *Container) bool {
 	if s.Podname != "" && s.Podname != c.Podname {
 		return false
 	}
+	if strings.HasSuffix(s.Name, "*") && strings.HasPrefix(c.Name, s.Name[:len(s.Name) - 1]) {
+		return true
```

Comment:
> I do not really want this patch to be included.
> This was more about what I needed to do to generate the profile rather than a polished patch to include.

Did you use local-gadget to inspect Inspektor Gadget?

In gadgettracermanager, `c.Name` does not contain the docker name but the container name as seen in the pod yaml:
```
# docker ps
5f1e29001f84   5185b96f0bec                  "/coredns -conf /etc…"   19 hours ago    Up 19 hours              k8s_coredns_coredns-565d847f94-vjl8g_kube-system_ec3920c7-2cac-4183-be79-124ea7488408_0
```
=> We see the docker name `k8s_coredns_coredns-565d847f94...`
```
# gadgettracermanager -dump
... Namespace:kube-system Podname:coredns-565d847f94-vjl8g Name:coredns
```
=> We see that `c.Name` = `coredns`